File: Ollimpiada/coockies.py

a = int(input())

def funk(n):
    d = {}
    counter = 0
    warning = 0
    for _ in range(n):
        b = input()
        if b not in d:
            d[b] = 1
        else:
            d[b] += 1
        counter +=1     
        
        if d[b] > 1 and (d[b] - 1 > counter - d[b]):
            warning +=1   
    return warning

print(funk(a))

                

File: Ollimpiada/coockies.cpp
#include <iostream>
#include <cstring>
#include <ctime>
#include <cstdlib>
#include <map>

using namespace std;


int main(){ 
    int size;
    cin >> size;
    map<string, int> wordMap;
    int counter = 0;
    int warning = 0;
    for (int i = 0;  i < size; i++){
        string name;
        cin >> name;
        if (wordMap.count(name) > 0){
            wordMap[name] +=1;
        }
        else{
            wordMap[name] = 1;
        }
        counter +=1;

        if (wordMap[name] > 1 && wordMap[name] -1 > counter - wordMap[name]){
            warning +=1;
        }
    } 
    cout <<"\n"<< warning<< endl;
}    
